specs/_features/maxeb_increase/capella.py

```diff
@@ -3767,14 +3761,37 @@ def is_fully_withdrawable_validator(validator: Validator, balance: Gwei, epoch:
         and balance > 0
     )
 
+
+def get_validator_effective_balance_ceiling(validator: Validator) -> Gwei:
+    if not has_compounding_withdrawal_credential(validator):
+        return MIN_ACTIVATION_BALANCE
+
+    custom_ceiling = Gwei(bytes_to_uint64(validator.withdrawal_credentials[1:3])) * EFFECTIVE_BALANCE_INCREMENT
+
+    # If set must not be higher than MAX_EFFECTIVE_BALANCE
+    if custom_ceiling > MIN_ACTIVATION_BALANCE:
+        return min(MAX_EFFECTIVE_BALANCE, custom_ceiling)
+
+    # If set must not be lower than MIN_ACTIVATION_BALANCE
```

Comment:
comment looks like it belongs on the other conditional ?

Reply:
not following what this second conditional does actually

Reply:
The branch below hits when `0 < custom_ceiling <= MIN_ACTIVATION_BALANCE`. We don’t allow it to be set to a value lower than `MIN_ACTIVATION_BALANCE` by having this check.

If `custom_ceiling == 0` then we consider it as unset and return `MAX_EFFECTIVE_BALANCE` — I am not sure about what to return in this case: `MIN_ACTIVATION_BALANCE` or `MAX_EFFECTIVE_BALANCE`. And decided to return the latter